### PR TITLE
4762:  Restructure urls and add capabilities filter ui tests

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/pages/solutions-list/context.js [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
+    supplierName: solution.supplier && solution.supplier.name,
+    capabilities: solution.capabilities && solution.capabilities.map(capability => capability.name),
+    isFoundation: solution.isFoundation,
+    viewSolutionUrl,
+  });
+});
 
 export const createSolutionListPageContext = ({
-  filterType, solutionListManifest, solutionsData,
+  filterType, solutionListManifest, solutionsData, query,
```
Comment (alibombaye) [inline app/pages/solutions-list/context.js]: can we name query to be something more specific like `selectedCapabilitiesQuery`

File: app/routes.js [hunk 2]
```diff
@@ -40,33 +40,43 @@ router.get('/solutions', (req, res) => {
...
-
-router.post('/solutions/custom', withCatch(async (req, res) => {
-  const capabilitiesSelected = req.body.capabilities;
-  const context = await getSolutionsForSelectedCapabilities({ capabilitiesSelected });
-  res.render('pages/solutions-list/template.njk', addConfig(context));
+router.get('/solutions/:filterType.:capabilities?', withCatch(async (req, res) => {
+  const { filterType, capabilities } = req.params;
+  if (filterType === 'capabilities-selector') {
+    if (capabilities) {
+      const formattedCapabilities = capabilities.split('+');
```
Comment (alibombaye) [inline app/routes.js]: can we move this line into `getSolutionsForSelectedCapabilities` function instead?

File: app/routes.js [hunk 3]
```diff
@@ -40,33 +40,43 @@ router.get('/solutions', (req, res) => {
...
+      const context = await getSolutionsForSelectedCapabilities({
+        capabilitiesSelected: formattedCapabilities,
+      });
+      logger.info(`navigating to capabilities-selector (with ${capabilities} selected) page`);
```
Comment (alibombaye) [inline app/routes.js]: Should this be navigating to the solutions-list page?

File: app/routes.js [hunk 4]
```diff
@@ -40,33 +40,43 @@ router.get('/solutions', (req, res) => {
...
+      res.render('pages/solutions-list/template.njk', addConfig(context));
+    } else {
+      const context = await getCapabilitiesContext();
+      logger.info('navigating to capabilities-selector page');
+      res.render('pages/capabilities-selector/template.njk', addConfig(context));
```
Comment (alibombaye) [inline app/routes.js]: Should we do a redirect here instead to the capability-selector route?

Comment (kat1906) [reply app/routes.js]: there is no capability-selector route now

File: app/routes.js (hunk 3 above)
Comment (kat1906) [reply app/routes.js]: yes

File: app/routes.js (hunk 2 above)
Comment (kat1906) [reply app/routes.js]: ok

File: app/routes.js (hunk 4 above)
Comment (alibombaye) [reply app/routes.js]: ah ok so this routes handles the capability selector page and the results page? I get it now 👍

Comment (kat1906) [reply app/routes.js]: yes it all goes through the :filterType route

**Comment by kat1906** [comment]
> Just some minor comments but otherwise this looks really good. Glad we put bit of extra thought into this.
> 
> However for a bonus point. Would it be possible to have a test to check for when multiple capabilities are selected and check that the url is as expected?

I forgot to add these! Done it now 😃

**Pull request merged**